sklearn/base.py

```diff
@@ -603,6 +604,45 @@ def fit_predict(self, X, y=None):
         return self.fit(X).predict(X)
 
 
+class OutlierRejectionMixin:
+    """Mixin class for all outlier detection resamplers in scikit-learn. Child
+    classes remove outliers from the dataset.
+    """
+    _estimator_type = "outlier_rejector"
+
+    def fit_resample(self, X, y, **kws):
```

Comment:
I think we should just be adding this to OutlierMixin rather than adding a new mixin and type.